### PR TITLE
Column Customization: Add slideout affordance and toggle button

### PR DESCRIPTION
**Pull request by JamesHollyer**
* Motivation for features / changes
This affordance will be used to house the column customization menu. For reference to where we are heading please see this branch(https://github.com/tensorflow/tensorboard/compare/master...JamesHollyer:tensorboard:EditColumnMenu).

* Technical description of changes
The slideout affordance is added to the mainViewComponent as it must be able to move outside the settings menu. When it is toggled we add the .slide-out-menu-expanded class which moves it to be in view. This with the transition property make the slide out animation.

* Screenshots of UI changes

### CODE REVIEW

**Comment by rileyajones** [comment]
This breaks the settings panel in dark mode
image

File: tensorboard/webapp/metrics/views/main_view/main_view_component.scss [hunk 1]
```diff
@@ -160,3 +161,18 @@ limitations under the License.
...
+  left: calc(100vw - 250px);
+  width: 200px;
+  background-color: white;
+  border: 1px solid mat.get-color-from-palette(mat.$gray-palette, 300);
+  transition: all 0.75s ease;
+}
+
+.slide-out-menu-expanded {
+  left: calc(100vw - 450px);
+}
```
Comment (rileyajones) [inline tensorboard/webapp/metrics/views/main_view/main_view_component.scss:178]: It's hard to capture but this existing leads to weird behavior when resizing the page.
Two things:
1) This doesn't need to be visible when not expanded.
2) There are a lot of calls to `calc` here ad I don't think you actually need any of them. You will also need to give `.split-content` `position: relative;` on line 75.
```suggestion
.slide-out-menu {
  @include tb-theme-background-prop(background-color, selected-button);
  position: absolute;
  height: 100%;
  right: 250px;
  width: 0;
  transition: all 0.75s ease;
  visibility: hidden;
}

.slide-out-menu-expanded {
  border-left: 1px solid mat.get-color-from-palette(mat.$gray-palette, 300);
  width: 200px;
  visibility: visible;
}
```

File: tensorboard/webapp/metrics/views/main_view/main_view_test.ts [hunk 2]
```diff
@@ -1722,6 +1723,51 @@ describe('metrics main view', () => {
...
+      const fixture = TestBed.createComponent(MainViewContainer);
+      fixture.detectChanges();
+
+      const slideoutMenu = fixture.debugElement.query(
+        By.css('.slide-out-menu')
+      );
+      expect(slideoutMenu).toBeFalsy();
+    });
+
+    it('renders slideout menu behind sidepanel when closed', () => {
```
[the remark this answers is not in the document]
Comment (JamesHollyer) [reply tensorboard/webapp/metrics/views/main_view/main_view_test.ts:1738]: Changed to non-expanded.

File: tensorboard/webapp/metrics/views/main_view/main_view_component.scss [hunk 3]
```diff
@@ -80,6 +80,7 @@ limitations under the License.
   overflow-x: hidden;
   overflow-y: auto;
   will-change: transform, scroll-position;
+  background-color: white;
```
[the remark this answers is not in the document]
Comment (JamesHollyer) [reply tensorboard/webapp/metrics/views/main_view/main_view_component.scss]: Wow! Good catch. Fixed.

File: tensorboard/webapp/metrics/views/main_view/main_view_component.scss (hunk 1 above)
Comment (JamesHollyer) [reply tensorboard/webapp/metrics/views/main_view/main_view_component.scss]: I changed this to not need calc and fixed the delayed moving issue you discussed.

Also reordered properties in alphabetic order.

**Review by JamesHollyer — COMMENTED** [review]
Thanks Riley.

File: tensorboard/webapp/metrics/views/main_view/main_view_component.scss [hunk 4]
```diff
@@ -160,3 +165,25 @@ limitations under the License.
     margin-right: 4px;
   }
 }
+
+.slide-out-menu {
+  background-color: mat.get-color-from-palette($tb-background, background);
+  height: 100%;
+  position: absolute;
+  right: 50px;
+  top: 50px;
```
Comment (rileyajones) [inline tensorboard/webapp/metrics/views/main_view/main_view_component.scss:174]: Where does this `top: 50px` come from? It looks like there is a gap
image

Comment (JamesHollyer) [reply tensorboard/webapp/metrics/views/main_view/main_view_component.scss]: So I have determined that it should be 49px. The regex filter bar is set to 48px with a bottom border of 1 px. The gap you have there is really big so not sure what how that happened but it looks good for me.

**Review by rileyajones — APPROVED** [review]

**Pull request merged**